### ISSUE TITLE
useFocusLock hook: Doesn't lock focus if last focusable element is disabled

### ISSUE THREAD

**Issue by YaroslavAntipov359** [cengage/react-magma#1272]
**Describe the Bug**
The useFocusLock hook is intended to lock focus within a specific component or element. However, when the last focusable element in the focus-locked area is disabled, focus can escape from the locked area, causing unexpected behavior.

**Steps to reproduce the behavior:**

1. Navigate to a component using the useFocusLock hook.
2. Disable the last focusable element within the focus-locked area.
3. Attempt to navigate through focusable elements using keyboard navigation (e.g., tab key).
4. Observe that focus is not contained within the designated focus-locked area.

**Expected Behavior**
The focus should remain within the specified focus-locked area, even when the last focusable element is disabled. The useFocusLock hook should correctly manage focus to ensure it does not escape the designated area, ensuring accessibility and a consistent user experience.

**Sandbox**
[Codesandbox with reproduced issue](https://codesandbox.io/p/sandbox/react-magma-example-forked-gnp4jr?file=%2Fexample.tsx%3A24%2C76&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvc7xzai00063b6m4igxrrrr%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvc7xzai00023b6maz1k08i3%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvc7xzai00033b6moocexh6z%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvc7xzai00053b6mshe98e3k%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvc7xzai00023b6maz1k08i3%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvc7xzai00013b6mku1hwgr2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clvc7yjrg00433b6m1be41h99%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A24%252C%2522startColumn%2522%253A76%252C%2522endLineNumber%2522%253A24%252C%2522endColumn%2522%253A76%257D%255D%252C%2522filepath%2522%253A%2522%252Fexample.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvc7xzai00023b6maz1k08i3%2522%252C%2522activeTabId%2522%253A%2522clvc7yjrg00433b6m1be41h99%2522%257D%252C%2522clvc7xzai00053b6mshe98e3k%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvc7xzai00043b6mmwxy8qb8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvc7xzai00053b6mshe98e3k%2522%252C%2522activeTabId%2522%253A%2522clvc7xzai00043b6mmwxy8qb8%2522%257D%252C%2522clvc7xzai00033b6moocexh6z%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvc7xzai00033b6moocexh6z%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Desktop:**
 - Mac OS
 - Chrome
 - 124

**Comment by silvalaura** [comment]
Hi @YaroslavAntipov359 . I can't reproduce this anymore with our latest version if I'm using an actual Modal instead of styled divs: https://codesandbox.io/p/sandbox/modal-with-last-item-disabled-n3xnm7?file=%2Fexample.tsx%3A13%2C1 

Can you confirm this is resolved for your use case? or, what is my example missing?

**Comment by YaroslavAntipov359** [comment]
Hi, @silvalaura.
I’ve updated my CodeSandbox to use the latest version of react-magma-dom@4.6.0, but I’m still able to reproduce the issue. In our case, the problem occurs with the useFocusLock hook, not the Modal component.

[Updated Codesandbox](https://codesandbox.io/p/sandbox/react-magma-example-forked-gnp4jr?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvc7xzai00063b6m4igxrrrr%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvc7xzai00023b6maz1k08i3%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvc7xzai00033b6moocexh6z%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvc7xzai00053b6mshe98e3k%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvc7xzai00023b6maz1k08i3%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvc7xzai00013b6mku1hwgr2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clvc7yjrg00433b6m1be41h99%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A24%252C%2522startColumn%2522%253A76%252C%2522endLineNumber%2522%253A24%252C%2522endColumn%2522%253A76%257D%255D%252C%2522filepath%2522%253A%2522%252Fexample.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvc7xzai00023b6maz1k08i3%2522%252C%2522activeTabId%2522%253A%2522clvc7yjrg00433b6m1be41h99%2522%257D%252C%2522clvc7xzai00053b6mshe98e3k%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvc7xzai00043b6mmwxy8qb8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvc7xzai00053b6mshe98e3k%2522%252C%2522activeTabId%2522%253A%2522clvc7xzai00043b6mmwxy8qb8%2522%257D%252C%2522clvc7xzai00033b6moocexh6z%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvc7xzai00033b6moocexh6z%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by silvalaura** [comment]
Thanks Yarik! We made a few updates to `useFocusLock` as of our latest dev version (4.7.0-next.12) but I am still able to see your issue within your CS with that version, so I will keep this ticket open to investigate.

**Comment by silvalaura** [comment]
Fixed in `4.7.0-next.18` ✅
